kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,307 @@
+        pollService = null;
+        notifyService.shutdownNow();
+        notifyService = null;
+        if (deploymentService != null) {
+            deploymentService.removeListener(this);
+            deploymentService = null;
+        }
+        if (consumer != null) {
+            consumer.close();
+            consumer = null;
+        }
+        initialized.set(false);
+    }
+
+    protected Consumer<String, byte[]> getKafkaConsumer() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, System.getProperty(
+                KAFKA_EXTENSION_PREFIX + "boopstrap.servers", "localhost:9092"));
+        // read only committed events
+        props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString().toLowerCase());
+        // do not automatically create topics by default
+        props.put(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG, Boolean.getBoolean(
+                KAFKA_EXTENSION_PREFIX + "auto.create.topics"));
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, System.getProperty(KAFKA_EXTENSION_PREFIX + "group.id",
+                "jbpm-consumer"));
+        return new KafkaConsumer<>(props, new StringDeserializer(), new ByteArrayDeserializer());
+    }
+
+    @Override
+    public void onDeploy(DeploymentEvent event) {
+        updateRegistration(event, this::updateTopics);
+    }
+
+    @Override
+    public void onUnDeploy(DeploymentEvent event) {
+        updateRegistration(event, this::removeTopics);
+    }
+
+    @Override
+    public void onActivate(DeploymentEvent event) {
+        updateRegistration(event, this::updateTopics);
+    }
+
+    @Override
+    public void onDeactivate(DeploymentEvent event) {
+        updateRegistration(event, this::removeTopics);
+    }
+
+    private void updateTopics(String deploymentId, ProcessDefinition processDefinition) {
+        for (SignalDesc signal : processDefinition.getSignalsMetadata()) {
+            topic2Signal.computeIfAbsent(topicFromSignal(signal), t -> new SignalInfo(deploymentId,
+                    signal));
+        }
+    }
+
+    private void removeTopics(String deploymentId, ProcessDefinition processDefinition) {
+        for (SignalDesc signal : processDefinition.getSignalsMetadata()) {
+            topic2Signal.remove(topicFromSignal(signal));
+        }
+    }
+
+    private void updateRegistration(DeploymentEvent event, BiConsumer<String, ProcessDefinition> updater) {
+        if (consumer != null) {
+            for (DeployedAsset asset : event.getDeployedUnit().getDeployedAssets()) {
+                updater.accept(event.getDeploymentId(), (ProcessDefinition) asset);
+            }
+            synchronized (consumer) {
+                consumer.subscribe(topic2Signal.keySet());
+            }
+        }
+    }
+
+    private String topicFromSignal(SignalDesc signal) {
+        return System.getProperty(TOPIC_PREFIX + signal.getName(), signal.getName());
```

Comment:
this is far to generic and it is not documented in the BAPL. 
signal names within process does not necessarily mean the same among process I would argue this is a good idea in the general case. 
While I can see the benefit (and indeed you can leave it like this). You would need to search first for more particular signal definition

The most particular case (this should be mandatory at least)
TOPIC_PREFIX.<deploymentId>.<processId>.signalName=signalName

more general (by deployment)
TOPIC_PREFIX.<deploymentId>.signalName = signalName

or your case (by entire server -> this is ok too)
TOPIC_PREFIX.signalName = signalName

Also requires to say that some signal is bounded. For instance is there is a signal registed by a process there is need to set an info kjar - process id - signal has been bounded to topic (whatever topic is mapped) through system property whaever.

The spec how to define the mapping should be part of the documentation of your BAPL as well.

This would add a lot of info regarding the mapping and allow diagnostics.

Reply:
Leaving at it is for now. Opened https://issues.redhat.com/browse/JBPM-9455 to allow fine-grainer approach.